### PR TITLE
Support for multi-channel files

### PR DESCRIPTION
**Pull request by Sidelobe**
Added support for multi-channel files. Did not have to do a lot - mainly adding tests to make sure it works. Currently testing writing 8-channel files for all formats and reading with 8-channel wav 24bit, 48k
Updated python script to generate headers as well.

### CODE REVIEW

File: AudioFile.cpp [hunk 1]
```diff
@@ -297,9 +297,9 @@ bool AudioFile<T>::decodeWaveFile (std::vector<uint8_t>& fileData)
     }
     
     // check the number of channels is mono or stereo
-    if (numChannels < 1 ||numChannels > 2)
+    if (numChannels < 1 || numChannels > 128)
     {
-        std::cout << "ERROR: this WAV file seems to be neither mono nor stereo (perhaps multi-track, or corrupted?)" << std::endl;
+        std::cout << "ERROR: this WAV file seems to be have an odd number of channels" << std::endl;
```
Comment (MatthieuHernandez) [inline AudioFile.cpp]: Why odd number ?
```suggestion
        std::cout << "ERROR: this WAV file have an invalid number of channels" << std::endl;
```

File: tests/AudioFileTests/FileWritingTests.cpp [hunk 2]
```diff
@@ -29,8 +29,16 @@ bool writeTestAudioFile (int numChannels, int sampleRate, int bitDepth, AudioFil
...
-    
-    std::string numChannelsAsString = numChannels == 1 ? "mono" : "stereo";
+
+    std::string numChannelsAsString;
+    if (numChannels == 1) {
+        numChannelsAsString = "mono";
+    } else if (numChannels == 2) {
+        numChannelsAsString = "stereo";
+    } else {
+        numChannelsAsString = std::to_string(numChannels) + "chan";
```
Comment (MatthieuHernandez) [inline tests/AudioFileTests/FileWritingTests.cpp]:
```suggestion
        numChannelsAsString = std::to_string(numChannels) + " channels";
```

**Review by MatthieuHernandez — APPROVED** [review]

File: AudioFile.cpp (hunk 1 above)
Comment (Sidelobe) [reply AudioFile.cpp]: Good point -- I probably meant "odd" as in "strange" -- but it's much clearer this way, thanks

**Comment by Sidelobe** [comment]
I've updated my branch to the newest develop, which in the meantime has switched to header-only .. tests are passing on my machine

**Comment by Sidelobe** [comment]
Any chance of this getting merged?

**Comment by adamstark** [comment]
Sorry I haven't been able to take a look at this yet - I will do that in the next few days, thanks for contributing! :)

Adam

**Review by adamstark — APPROVED** [review]

**Comment by adamstark** [comment]
Got stuck into it right now - I've just merged it in. Thanks again! :)

I'll need to catch up on this to get it pushed to master, will try and do that over the next couple of days as i need to review some other new features on develop.

Adam

**Comment by adamstark** [comment]
BTW, what can I use to test / open / edit 8 channel WAV files? This was one of the big blockers for me :)

**Comment by Sidelobe** [comment]
Awesome, thanks Adam!

To quickly create/edit multichannel wave files, I use Audacity -- ugly interface, but extremely powerful :-)
https://www.audacityteam.org/

**Pull request merged**